sdk/src/test/java/com/finbourne/lusid/integration/TransactionsTests.java

```diff
@@ -0,0 +1,204 @@
+        String originalPortfolioId = String.format("Id-%s", uuid);
+        CreateTransactionPortfolioRequest request = new CreateTransactionPortfolioRequest()
+                .displayName(String.format("Portfolio-%s", uuid))
+                .code(originalPortfolioId)
+                .baseCurrency("GBP")
+                .created(effectiveDate);
+
+        //  create portfolio
+        Portfolio portfolio = this.transactionPortfoliosApi.createPortfolio(scope, request);
+
+        assertEquals(portfolio.getId().getCode(), originalPortfolioId);
+
+        String portfolioId = portfolio.getId().getCode();
+
+        //  create the transaction
+        TransactionRequest transaction = new TransactionRequest()
+                .transactionId(UUID.randomUUID().toString())
+                .type("Buy")
+
+                //  instruments must already exist in LUSID and have a valid LUSID instrument id
+                .instrumentUid(this.instrumentIds.get(0))
+                .totalConsideration(new CurrencyAndAmount().currency("GBP").amount(1230.0))
+                .transactionDate(effectiveDate)
+                .settlementDate(effectiveDate)
+                .units(100.0)
+                .transactionPrice(new TransactionPrice().price(12.3))
+                .source(TransactionRequest.SourceEnum.CLIENT);
+
+        //  add the trade
+        this.transactionPortfoliosApi.upsertTransactions(scope, portfolioId, new ArrayList<>(Arrays.asList(transaction)));
+
+        //  get the trade
+        VersionedResourceListOfTransaction transactions = this.transactionPortfoliosApi.getTransactions(scope,
+                portfolioId, null, null, null, null, null, null, null, null);
+
+        assertEquals(1, transactions.getValues().size());
+        assertEquals(transaction.getTransactionId(), transactions.getValues().get(0).getTransactionId());
+    }
+    
+    @Test
+    public void load_cash() throws ApiException {
+        String uuid = UUID.randomUUID().toString();
+        String scope = "finbourne";
+        OffsetDateTime effectiveDate = OffsetDateTime.of(2018, 1, 1, 0, 0, 0, 0, ZoneOffset.UTC);
+
+        //  portfolio request
+        String originalPortfolioId = String.format("Id-%s", uuid);
+        CreateTransactionPortfolioRequest request = new CreateTransactionPortfolioRequest()
+                .displayName(String.format("Portfolio-%s", uuid))
+                .code(originalPortfolioId)
+                .baseCurrency("GBP")
+                .created(effectiveDate);
+
+        //  create portfolio
+        Portfolio portfolio = this.transactionPortfoliosApi.createPortfolio(scope, request);
+
+        assertEquals(portfolio.getId().getCode(), originalPortfolioId);
+
+        String portfolioId = portfolio.getId().getCode();
+
+        //  create the transaction
+        TransactionRequest transaction = new TransactionRequest()
+                .transactionId(UUID.randomUUID().toString())
+                .type("Buy")
```

Comment:
Probably best to use "FundsIn" transaction code, then you only need to specify Units not total consideration.  We should make TotalConsideration nullable, but in the meantime make it zero